src/middleware/error-handler.ts

```diff
@@ -0,0 +1,75 @@
+import { ErrorRequestHandler } from 'express';
+
+import { CustomErrorCodes } from '../enums';
+import {
+  ValidationIssues,
+  RequestValidationError,
+  TaskStartFinishTimeError,
+  TaskTimelineError
+} from '../errors/error';
+import httpStatus from 'http-status';
+
+interface ReqValidationErrRes {
+  success: false;
+  message: string;
+  code: CustomErrorCodes.BAD_REQUEST;
+  errors: ValidationIssues;
+}
+
+interface ServerErrRes {
+  success: false;
+  message: string;
+  code: CustomErrorCodes.SERVER_ERROR;
+}
+
+interface ReqNotProcessedErrRes {
+  success: false;
+  message: string;
+  code: CustomErrorCodes.NOT_PROCESSED;
+  errors: string[];
+}
+
+type ErrRes = ReqValidationErrRes | ServerErrRes | ReqNotProcessedErrRes;
+
+export const errorHandler: ErrorRequestHandler<unknown, ErrRes> = (err, _req, res, _next) => {
+  try {
+    switch (true) {
+      case err instanceof RequestValidationError:
+        console.error(err.toString(), '\n', JSON.stringify(err));
```

Comment:
I thought you had a custom logger. It is advised to limit the use of console.log in application codes